### PR TITLE
Use __dict__ or hasattr instead of dir.

### PR DESCRIPTION
**Pull request by atombrella**
##### SUMMARY

This looks like legacy to me.

##### ISSUE TYPE
- Bugfix Pull Request

### CODE REVIEW

**Comment by atombrella** [comment]
Tests don't run because of some Shippable issues?

**Comment by Fred-sun** [comment]
@atombrella Could you help resolve the CI fail? Thank you very much!

File: contrib/inventory/vmware.py [hunk 1]
```diff
@@ -185,7 +185,7 @@ def _get_obj_info(self, obj, depth=99, seen=None):
             if depth <= 0:
                 return obj_unicode
             d = {}
-            for attr in dir(obj):
+            for attr in obj.__dict__:
```
Comment (atombrella) [inline contrib/inventory/vmware.py:188]: I'm a little bit in doubt about these. `dir` returns more items than `__dict__`.

File: lib/ansible/cli/scripts/ansible_cli_stub.py [hunk 2]
```diff
@@ -92,7 +92,7 @@ def error(self, msg, wrap_text=None):
             mycli = getattr(__import__("ansible.cli.%s" % sub, fromlist=[myclass]), myclass)
         except ImportError as e:
             # ImportError members have changed in py3
-            if 'msg' in dir(e):
+            if hasattr(e, 'msg'):
```
Comment (atombrella) [inline lib/ansible/cli/scripts/ansible_cli_stub.py:95]: This is a definitely a sane substitution. This is what initially caught my attention.

**Comment by Fred-sun** [comment]
ready_for_review

**Comment by Fred-sun** [comment]
@gavinfish @haiyuazhang

**Comment by Fred-sun** [comment]
@atombrella Thank you for taking the time to contribute to this PR. We will transfer ansible 's azure module related Issue and PR to azure collection (https://github.com/ansible-collections/azure/pulls), can you transfer the Issue to azure collection repo?

**Comment by atombrella** [comment]
I'm closing in favor of the new repo.

**Pull request closed without merging**